force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -0,0 +1,67 @@
+public without sharing class TaskTriggerHandler {
+  public static void updateRelatedAccounts(List<Task> tasks) {
+    Map<Id, Account> accountMap = countSpecificTasks(
+      getUnitedStatesProspectAccounts(tasks)
+    );
+    update accountMap.values();
+  }
+
+  public static List<Account> getUnitedStatesProspectAccounts(
+    List<Task> tasks
+  ) {
+    Set<Id> accIds = new Set<Id>();
+    for (Task task : tasks) {
+      accIds.add(task.whatId);
```

Comment:
I see what you mean. 

If we use it like so, then accIds is not the correct variable name anymore but should be called WhatIds

And yes, we will use a query each time a task is inserted if we don't filter whatIds.

I just wanted to make the point that the query will work with unfiltered Ids as well. 

For the particular case, we should check if it is an accountId. Please find another documented way to check the id prefix without hardcoding '001' :)

Reply:
I think that with the new implementation, I should do the check in the method which is called by the Trigger so in updateRelatedAccounts.
![check](https://user-images.githubusercontent.com/77618420/217207387-f7319e67-e127-4260-864a-5e711e5e1008.jpg)

If my task is related to an Account, I should have an AccountId, right? At this point, wouldn't it be easier to check directly on the AccountId and not on WhatId? If AccountId is null, it means that my Task is not related to any account, so I need to stop the execution of my program. Maybe checking with 'task.accoundId != null' is a bit basic but works. What do you think?

Reply:
Or I can do this:
![image](https://user-images.githubusercontent.com/77618420/217220072-955ec40c-a71d-45e7-9bbd-21fbffc9f3ec.png)

But... what is the difference between checking on AccountId or checking on WhatId in this case? (When I need only the Tasks related to Accounts. I'm not talking about other cases, when the Tasks are related to other objects)